src/components/dapps/Message.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <div
+    class="cursor-default my-2 py-1 px-2 dark:text-white bg-gray-100 hover:bg-gray-200 dark:bg-darkGray-800 dark:hover:bg-darkGray-700 rounded-sm"
+  >
+    <div class="tooltip relative">
+      <div class="flex">
+        <div
+          class="font-semibold"
+          :class="[message.isConstructor ? 'text-blue-400' : 'text-yellow-400']"
+        >
+          {{ message.identifier }}
+        </div>
+        <div>{{ `(${argsString})${returnTypeString}` }}</div>
+      </div>
+      <!-- Tooltip -->
+      <span
+        class="tooltip-shadow dark:bg-darkGray-900 hidden absolute z-10 top-6 left-5 text-s dark:text-white bg-gray-800 rounded-md"
+      >
+        <div>
+          <div
+            class="flex bg-gray-100 dark:bg-darkGray-800 px-2 py-1 rounded-t-md"
+          >
+            <div
+              class="font-semibold"
+              :class="[
+                message.isConstructor ? 'text-blue-400' : 'text-yellow-400',
+              ]"
+            >
+              {{ message.identifier }}
+            </div>
+            <div>{{ `(${argsString})${returnTypeString}` }}</div>
+          </div>
+          <div
+            class="leading-7 px-2 py-2 float-left w-96 bg-gray-200 dark:bg-darkGray-900"
+          >
+            {{ docsString }}
+          </div>
+        </div>
+      </span>
+    </div>
+  </div>
+</template>
+<script lang="ts">
+// import useAbi from '@/hooks/useAbi';
+import { PropType, defineComponent } from 'vue';
+import type { AbiParam } from '@polkadot/api-contract/types';
+import type { TypeDef } from '@polkadot/types/create/types';
+interface MessageType {
```

Comment:
The same type is being repeated on `ContractInfo` component https://github.com/PlasmNetwork/plasm-apps-vue/pull/28/files#diff-797a2b6e48f6a83f652ea849b20fbb8f19a7e18c4f9c18484d433520ab983474R16
If we are using this in multiple places, please move this to a separate typescript file.